### ISSUE TITLE
Strict Mode: Document behavior of setState callbacks

### ISSUE THREAD

**Issue by fefrei** [reactjs/react.dev#6559]
As per [the documentation](https://react.dev/reference/react/StrictMode), Strict Mode has three effects:

> - Your components will [re-render an extra time](https://react.dev/reference/react/StrictMode#fixing-bugs-found-by-double-rendering-in-development) to find bugs caused by impure rendering.
> - Your components will [re-run Effects an extra time](https://react.dev/reference/react/StrictMode#fixing-bugs-found-by-re-running-effects-in-development) to find bugs caused by missing Effect cleanup.
> - Your components will [be checked for usage of deprecated APIs.](https://react.dev/reference/react/StrictMode#fixing-deprecation-warnings-enabled-by-strict-mode)

As per [his issue comment](https://github.com/facebook/react/issues/12856#issuecomment-390206425), “setState updaters will run twice in strict mode in development”. I think this behavior should be documented on the Strict Mode page.

**Comment by rickhanlonii** [comment]
Yes, this is documented on that same page [here](https://react.dev/reference/react/StrictMode#fixing-bugs-found-by-double-rendering-in-development) and in the docs for setState [here](https://react.dev/reference/react/useState#my-initializer-or-updater-function-runs-twice).

**Comment by fefrei** [comment]
I see, looks like I overread them, thanks. There might still be room for improvement, though: The list I quoted above is titled “Strict Mode enables the following development-only behaviors”, making it sound (to me) like it's an exhaustive lists of what strict mode does. This is what confused me – I saw some weird behavior due to a bug in my code, wondered whether this might be triggered do to the setState updater being run twice, saw the (exhaustive‽) list above, and discarded that theory. Maybe this list can be completed, or re-worded as just an example of the types of changes strict mode enables?
